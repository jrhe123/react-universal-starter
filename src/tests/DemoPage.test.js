import React from 'react';
import expect from 'expect';
import { shallow, mount } from 'enzyme';
import { DemoPage } from '../shared/demo/DemoPage';
import renderer from 'react-test-renderer';

describe('Component: test message here', () => {

    it('does all the things', () => {
        
        expect(true).toEqual(true);
    })
})

describe('Component: test message here', () => {

    it('render page', () => {
        
        const wrapper = shallow(<DemoPage />);
        const countState = wrapper.state().count;
        expect(countState).toEqual(0);
    })

    it('render page', () => {
        
        const wrapper = shallow(<DemoPage />);
        const text = wrapper.find('h1').text();
        expect(text).toEqual('Count: 0');
    })

    it('render page', () => {
        
        const wrapper = shallow(<DemoPage />);
        const incrementBtn = wrapper.find('button');
        incrementBtn.simulate('click');

        const text = wrapper.find('h1').text();
        expect(text).toEqual('Count: 1');
    })

    it('test snap shot', () => {
        
        const tree = renderer.create(<DemoPage />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('shallow vs mount', () => {
        
        const shallowWrapper = shallow(<DemoPage />);   // only one level of component
        const mountWrapper = mount(<DemoPage />);   // trees of components

        console.log('shallowWrapper: ', shallowWrapper.debug());
        console.log('mountWrapper: ', mountWrapper.debug());
    })

})