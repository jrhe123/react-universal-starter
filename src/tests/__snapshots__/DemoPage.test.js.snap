// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: test message here test snap shot 1`] = `
<div>
  <h1>
    Count: 
    0
  </h1>
  <button
    onClick={[Function]}
  >
    click
  </button>
</div>
`;
